scripts/cat/status.py

```diff
@@ -0,0 +1,639 @@
+        elif self.is_former_clancat() and not (
+            self.group and self.group.is_afterlife()
+        ):
+            new_rank = self.find_prior_clan_rank()
+            # we don't need to change leaders and deps if they're going to an afterlife
+            if (
+                new_rank in [CatRank.LEADER, CatRank.DEPUTY]
+                and not new_group.is_afterlife()
+            ):
+                new_rank = self.get_rank_from_age(age)
+        else:
+            new_rank = self.rank
+
+        if new_group.is_any_clan_group() and not new_rank.is_any_clancat_rank():
+            new_rank = self.get_rank_from_age(age)
+
+        self._modify_group(
+            new_rank=new_rank,
+            standing_with_past_group=standing_with_past_group,
+            new_group=new_group,
+        )
+
+    def send_to_afterlife(self, target: CatGroup = None):
+        """
+        Changes a cat's group into the appropriate afterlife
+        :param target: Use this to specify a certain afterlife, if unused a clancat (or a former clancat) will match
+        their guide's afterlife, while an outsider will go to the unknown residence.
+        """
+        # if we have a specific afterlife to send them to
+        if target:
+            self.add_to_group(
+                new_group=target,
+            )
+            return
+
+        # if we have an outsider who has never been a clancat, they go to the unknown residence
+        if self.is_outsider() and not self.is_former_clancat():
+            self.add_to_group(new_group=CatGroup.UNKNOWN_RESIDENCE)
+            return
+
+        # meanwhile clan cats go wherever their guide points them
+        if game.clan:
+            self.add_to_group(new_group=game.clan.instructor.status.group)
+        else:
+            self.add_to_group(new_group=CatGroup.STAR_CLAN)
+
+    def change_rank(self, new_rank: CatRank):
+        """
+        Changes the cats rank to the new_rank. Generally you shouldn't use just this to change a cat's rank!
+        cat.rank_change() should typically be called instead, since it will handle mentor switches and other complex
+        changes.
+        """
+        # checks that we don't add a duplicate group/rank pairing
+        if self.group_history:
+            last_entry = self.group_history[-1]
+            if last_entry["group"] == self.group and last_entry["rank"] == new_rank:
+                return
+
+        self.group_history.append(
+            {"group": self.group, "rank": new_rank, "moons_as": 0}
+        )
+
+    def change_group_nearness(self, group: CatGroup):
+        """
+        Flips the "near" bool of the given group.
+        """
+        for entry in self.standing_history:
+            if entry.get("group") == group:
+                if entry.get("near"):
+                    entry["near"] = False
+                else:
+                    entry["near"] = True
```

Comment:
```suggestion
                if entry.get("near"):
                    entry["near"] = not entry.get("near", False)
```
Also, what cases are there groups that don't have a `near` value? Is that Playerclan?

Reply:
There's actually no case where they can't have a near, so I guess it makes more sense to let it crash if that's missing somehow, there would def be something wrong going on.